File: app/src/main/java/ru/p3tr0vich/fuel/ApplicationFuel.java
package ru.p3tr0vich.fuel;

import android.annotation.SuppressLint;
import android.app.Application;
import android.content.Context;
import android.support.annotation.NonNull;

import ru.p3tr0vich.fuel.utils.UtilsLog;

public class ApplicationFuel extends Application {

    @SuppressLint("StaticFieldLeak")
    private static Context sContext;

    @Override
    public void onCreate() {
        super.onCreate();
        sContext = getApplicationContext();

        UtilsLog.d("ApplicationFuel", "**********************************************************");
        UtilsLog.d("ApplicationFuel", "onCreate");

        ContentObserverService.start(this);
    }

    @Override
    public void onTerminate() {
        UtilsLog.d("ApplicationFuel", "onTerminate");
        super.onTerminate();
    }

    @NonNull
    public static Context getContext() {
        if (sContext == null) throw new AssertionError("Application context is null");
        return sContext;
    }
}

File: app/src/main/kotlin/ru/p3tr0vich/fuel/ApplicationFuel.kt
package ru.p3tr0vich.fuel

import android.annotation.SuppressLint
import android.app.Application
import android.content.Context

import ru.p3tr0vich.fuel.utils.UtilsLog

class ApplicationFuel : Application() {

    override fun onCreate() {
        super.onCreate()

        sContext = applicationContext

        UtilsLog.d("ApplicationFuel", ">>> onCreate *************************************")

        ContentObserverService.start(this)
    }

    override fun onTerminate() {
        UtilsLog.d("ApplicationFuel", ">>> onTerminate **********************************")
        super.onTerminate()
    }

    companion object {

        @SuppressLint("StaticFieldLeak")
        private var sContext: Context? = null

        val context: Context
            get() {
                if (sContext == null) {
                    throw AssertionError("Application context is null")
                }

                return sContext!!
            }
    }
}